src/components/checkbox/Checkbox.tsx

```diff
@@ -1,40 +1,57 @@
-import React, { InputHTMLAttributes, useState } from 'react';
+import React, { ChangeEvent, InputHTMLAttributes, useId, useState } from 'react';
 import styled from '@emotion/styled';
 
 import IconCheckbox from '../icon/IconCheckbox';
 
 interface Props extends InputHTMLAttributes<HTMLInputElement> {
-  onCheck?: () => void;
-  onUncheck?: () => void;
+  onToggle?: (e: ChangeEvent<HTMLInputElement>) => void;
   defaultChecked?: boolean;
 }
 
-const Checkbox = ({ onCheck, onUncheck, defaultChecked = false }: Props) => {
+const Checkbox = ({ onToggle, defaultChecked = false }: Props) => {
   const [checked, setChecked] = useState<boolean>(defaultChecked);
-
-  const onToggle = () => {
-    setChecked((prev) => !prev);
-    if (!checked && onCheck) {
-      onCheck();
-      return;
-    }
-    if (onUncheck) {
-      onUncheck();
-    }
-  };
+  const id = useId();
 
   return (
-    <CheckboxWrapper onClick={onToggle} role="checkbox" aria-checked={checked}>
-      <IconCheckbox isChecked={checked} />
+    <CheckboxWrapper>
+      <StyledInput
+        type="checkbox"
+        onChange={(e) => {
+          setChecked(!checked);
```

Comment:
```suggestion
          setChecked(prev => !prev);
```

이렇게 이전 값을 보장하는 건 어떨까요 !?

Reply:
useToggle 사용하는 코드로 변경됐습니다!